packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -406,6 +405,47 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
     `, [userId]);
   }
 
+  async getDialogueRecommendedUsers(userId: string, upvotedUsers:UpvotedUser[], limit = 100): Promise<DbUser[]> {
+    const upvotedUserIds = upvotedUsers.map(user => user._id);
+
+    return this.any(`
+      SELECT u.*
+      FROM unnest($2::text[]) AS uv(_id)
+      INNER JOIN "Users" AS u ON uv._id = u._id
+      WHERE
+        -- Exclude users that the current user has already checked
+        (
+          SELECT COUNT(*)
+          FROM "DialogueChecks"
+          WHERE "userId" = $1 AND "targetUserId" = uv._id AND "checked" = TRUE
```

Comment:
always prefer `IS TRUE` to `= TRUE`; regular boolean equality comparisons fall over on null values.